### PR TITLE
Generate names for options

### PR DESCRIPTION
**Pull request by abhijitb**
## Proposed changes

In the AI site generation interface we will keep showing "Version 1", "Version 2", etc. but when the theme name and slug are generated we will use title + number like: "bean-there-cafe-3" and "Bean There Cafe 3".
If the user has chosen a name for a version we use that as title and slug for theme.
Note: This works only for favorited versions in the interface 
Sone examples :

**User chosen title**
Before :
    slug : version-3 => title : final-version
After :
    slug : version-3 => title : final version

**Default case 1**
Before :
slug : version-1 => title : version-1
After : 
slug : bean-there-cafe-1 => title :  Bean There Cafe 1 

**Default Case 2**
Before :
    slug : version-1-copy-copy => title : version 1 (copy) (copy)
After :
    slug : bean-there-cafe-1-copy-copy => title : Bean There Cafe 1 copy copy

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [x] Linting and tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: includes/Services/SiteGenService.php [hunk 1]
```diff
@@ -125,12 +125,48 @@ public static function instantiate_site_meta( $site_info, $identifier, $skip_cac
...
+	 * @return array
+	 */
+	public static function update_info_for_child_theme( $site_config, $data, $homepage_index = '' ) {
+		// Fetch title and slug for the child theme
+		$theme_name = \remove_accents( $site_config['site_title'] );
+		$theme_slug = \sanitize_title_with_dashes( $theme_name );
+
+		$homepage_slug              = $data['slug'];
+		$title                      = str_replace( ' ', '-', str_replace( array( '(', ')' ), '', $data['title'] ) );
+		$homepage_title_with_dashes = strtolower( str_replace( ' ', '-', $title ) );
```
Comment (officiallygod) [inline includes/Services/SiteGenService.php:143]: Doesn't the $title already replace '-' with ' '?

**Comment by officiallygod** [comment]
Rest looks good, Need Intensive QA on this with all cases too.

File: includes/Services/SiteGenService.php (hunk 1 above)
Comment (abhijitb) [reply includes/Services/SiteGenService.php:143]: fixed ... removed this variable $homepage_title_with_dashes and using $title itself

**Review by officiallygod — APPROVED** [review]
Looks Good for QA

**Pull request merged**